pkg/infra/metrics/metrics.go

```diff
@@ -104,6 +105,9 @@ var (
 
 	// MPublicDashboardRequestCount is a metric counter for public dashboards requests
 	MPublicDashboardRequestCount prometheus.Counter
+
+	// MPublicDashboardDatasourceQuerySuccess is a metric counter for successful queries labelled by datasource
+	MPublicDashboardDatasourceQuerySuccess *prometheus.CounterVec
```

Comment:
taking into account you're not only tracking success but also failures in the query, is MPublicDashboardDatasourceQuery**Success** the correct name?

Reply:
Maybe? `MPublicDashboardDatasourceQuerySuccessStatus` ? Its either successful or its not. Naming is hard.